arithmetization/src/main/java/net/consensys/linea/zktracer/types/TransactionUtils.java

```diff
@@ -24,10 +24,10 @@ public static long getChainIdFromTransaction(final Transaction tx) {
         return tx.getChainId().get().longValueExact();
       }
       case FRONTIER -> {
-        // when ChainId != 0, V = 2 * ChainID + 35 or V = 2 * ChainId + 36;
+        // If chainId is specified, V = 2 * ChainID + 35 or V = 2 * ChainId + 36;
         final long V = tx.getV().longValueExact();
         if (V == 27 || V == 28) {
-          return 0;
+          throw new IllegalArgumentException("ChainId not specified in transaction");
```

Comment:
What is the purpose of throwing an exception ? To print a message ? I see that we catch it and resume execution.

Reply:
we call this functiun in `phaseBeta` of `RLP_TXN` and here we'll be happy to catch exception if needed

Reply:
> it's if some transaction defines a chainId which does not coincide with that of Linea mainnet / testnet. 

Those transactions are supposed to be blocked by the sequencer.